Board.java

```diff
@@ -0,0 +1,634 @@
+            new BoardSquare(),
+    }, {
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 2),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 2),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+    }, {
+            new BoardSquare(2, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 2),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(2, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 2),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(2, 1),
+    }, {
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 2),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(2, 1),
+            new BoardSquare(),
+            new BoardSquare(2, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 2),
+            new BoardSquare(),
+            new BoardSquare(),
+    }, {
+            new BoardSquare(),
+            new BoardSquare(1, 2),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(3, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(3, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 2),
+            new BoardSquare(),
+    }, {
+            new BoardSquare(1, 3),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(2, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 3),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(2, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 3),
+    }
+    };
+    private static Board SingleBoard = null;
+
+    private Board() {}
+
+    public static Board getBoard() {
+        if(SingleBoard == null) {
+            SingleBoard = new Board();
+        }
+        return SingleBoard;
+    }
+
+    public Tile[][] getTiles() {
+        Tile[][] tilesBoard = new Tile[BOARD_SIZE][BOARD_SIZE];
+
+        for (int i = 0; i < board[0].length; i++) {
```

Comment:
it's better to use java library when possible.. check system arraycopy

Reply:
i checked now about the arraycopy function, the thing here is that i create a 2d array of type Tile but the type for the board is BoardSquare